opentelemetry-sdk/src/testing/metrics/in_memory_exporter.rs

```diff
@@ -46,7 +47,8 @@ use std::sync::{Arc, Mutex};
 ///  // Create and record metrics using the MeterProvider
 ///  let meter = meter_provider.meter(std::borrow::Cow::Borrowed("example"));
 ///  let counter = meter.u64_counter("my_counter").init();
-///  counter.add(1, &[KeyValue::new("key", "value")]);
+///  let attributes = AttributeSet::from(&[KeyValue::new("key", "value")]);
```

Comment:
i am not sure if we should change all examples to use AttributeSet.. maybe show it only where it gives some perf benefits.

Reply:
These examples were changed prior to a backward compatible solution being found, thus I had to update all the doctests to use precreated attributes for CI to pass.